src/org/labkey/filetransfer/FileTransferController.java

```diff
@@ -119,10 +151,362 @@ public URLHelper getSuccessURL(FileTransferConfigForm form)
+     * authorization UI.  This redirect contains a parameter indicating the action to return to when authentication is
+     * complete.
+     */
+    @CSRF
+    @RequiresPermission(ReadPermission.class)
+    public class AuthAction extends SimpleViewAction<TransferSelectionForm>
+    {
+        @Override
+        public NavTree appendNavTrail(NavTree root)
+        {
+            return null;
+        }
+
+        @Override
+        public ModelAndView getView(TransferSelectionForm form, BindException errors) throws Exception
+        {
+            HttpSession session = getViewContext().getRequest().getSession();
+            session.setAttribute(DATA_REGION_SELECTION_KEY, form.getDataRegionSelectionKey());
+            session.setAttribute(FILE_TRANSFER_CONTAINER, getContainer().getId());
+            session.setAttribute(FileTransferManager.WEB_PART_ID_SESSION_KEY, form.getWebPartId());
+            session.setAttribute(FileTransferManager.RETURN_URL_SESSION_KEY, form.getReturnUrl());
+            FileTransferProvider provider = FileTransferManager.get().getProvider(getViewContext());
+            if (provider != null)
+            {
+                OAuth2Authenticator authenticator = provider.getAuthenticator(getContainer(), getUser());
+                throw new RedirectException(authenticator.getAuthorizationUrl());
+            }
+            else
+            {
+                errors.reject("No File Transfer Provider defined for this web part");
+                return null;
+            }
+        }
+    }
+
+    public static class TransferSelectionForm extends ReturnUrlForm
+    {
+        private String dataRegionSelectionKey;
+        private Integer webPartId;
+
+        public String getDataRegionSelectionKey()
+        {
+            return dataRegionSelectionKey;
+        }
+
+        public void setDataRegionSelectionKey(String dataRegionSelectionKey)
+        {
+            this.dataRegionSelectionKey = dataRegionSelectionKey;
+        }
+
+        public Integer getWebPartId()
+        {
+            return webPartId;
+        }
+
+        public void setWebPartId(Integer webPartId)
+        {
+            this.webPartId = webPartId;
+        }
+    }
+
+    /**
+     * This action is the target of the authorization action from the file transfer provider.  If authorization has
+     * been granted, the code provided from that authorization will be used to retrieve credentials to be used in
+     * initiating transfer requests.  In any case, this action redirects to a page where we give feedback to the user
+     * and display items for next steps, which are either to return to the page where the initial selection of files was
+     * made, choose a destination endpoint, or initiate a transfer of the selected files.
+     */
+    @CSRF
+    @RequiresPermission(ReadPermission.class)
+    public class TokensAction extends RedirectAction<AuthForm>
+    {
+        Boolean authorized = true;
```

Comment:
The default for this param is TRUE? I would have expected it to be NULL or FALSE, but I could be wrong.